pkg/cmd/operator/mustgather.go

```diff
@@ -0,0 +1,290 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/scylladb/scylla-operator/pkg/gather/collect"
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	"github.com/scylladb/scylla-operator/pkg/signals"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/discovery"
+	"k8s.io/klog/v2"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	mustGatherLongDescription = templates.LongDesc(`
+		must-gather collects resource dumps.
+	
+		This command is experimental and a subject to change without notice.
+	`)
+
+	mustGatherExample = templates.Examples(`
+		# Collect archive of all resources related to scyllacluster and its APIs.
+		scylla-operator must-gather
+		
+		# Collect archive of all resources present in the Kubernetes cluster.
+		scylla-operator must-gather --all-resources
+	`)
+)
+
+type MustGatherOptions struct {
+	*GatherBaseOptions
+
+	AllResources bool
+}
+
+func NewMustGatherOptions(streams genericclioptions.IOStreams) *MustGatherOptions {
+	return &MustGatherOptions{
+		GatherBaseOptions: NewGatherBaseOptions("scylla-operator-must-gather", true),
+		AllResources:      false,
+	}
+}
+
+func (o *MustGatherOptions) AddFlags(flagset *pflag.FlagSet) {
+	o.GatherBaseOptions.AddFlags(flagset)
+
+	flagset.BoolVarP(&o.AllResources, "all-resources", "", o.AllResources, "Gather will discover preferred API resources from the apiserver.")
+}
+
+func NewMustGatherCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewMustGatherOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:     "must-gather",
+		Short:   "Run the scylla must-gather.",
+		Long:    mustGatherLongDescription,
+		Example: mustGatherExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run(streams, cmd)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+		ValidArgs: []string{},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	o.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (o *MustGatherOptions) Validate() error {
+	var errs []error
+
+	errs = append(errs, o.GatherBaseOptions.Validate())
+
+	return utilerrors.NewAggregate(errs)
```

Comment:
Can we get rid of this? [errors.Join ](https://pkg.go.dev/errors#Join)was added to stdlib in 1.20.